RealmSwift/ThreadSafeReference.swift

```diff
@@ -122,6 +122,76 @@ public protocol ThreadConfined {
     }
 }
 
+// MARK: ThreadSafe propertyWrapper
+
+/**
+    A property wrapper type that may be passed between threads.
+
+    A `@ThreadSafe` property contains a thread-safe reference to the underlying wrapped value.
+    This reference is resolved to the thread on which the wrapped value is accessed. A new thread
+    safe reference is created each time the property is accessed.
+
+ - warning: This property wrapper should not be used for properties on long lived objects.
+            `@ThreadSafe` properties contain a `ThreadSafeReference` which
+            can pin the source version of the Realm in use. This means that this property
+            wrapper is **better suited for function arguments and local variables**
+            **that get captured by an aynchronously dispatched block.**
+
+ - see: `ThreadSafeReference`
+ - see: `ThreadConfined`
+*/
+@propertyWrapper public class ThreadSafe<T: ThreadConfined> {
+    private var threadSafeReference: ThreadSafeReference<T>?
+    private var rlmConfiguration: RLMRealmConfiguration?
+
+    /// :nodoc:
+    public var wrappedValue: T? {
+        get {
+            guard let threadSafeReference = threadSafeReference,
+                  let rlmConfig = rlmConfiguration else { return nil }
+            do {
+                let rlmRealm = try RLMRealm(configuration: rlmConfig)
+                let realm = Realm(rlmRealm)
+                guard let value = threadSafeReference.resolve(in: realm) else {
+                    self.threadSafeReference = nil
+                    return nil
+                }
+                self.threadSafeReference = ThreadSafeReference(to: value)
+                return value
+            // FIXME: wrappedValue should throw
+            // As of Swift 5.5 property wrappers can't have throwing accessors.
+            } catch let error as NSError {
+                throwRealmException(error.localizedDescription)
+            }
+        }
+        set {
+            guard let newValue = newValue else {
+                threadSafeReference = nil
+                return
+            }
+            guard let rlmConfiguration = newValue.realm?.rlmRealm.configuration else {
+                throwRealmException("Only managed objects may be wrapped as thread safe.")
+            }
+            self.rlmConfiguration = rlmConfiguration
+            threadSafeReference = ThreadSafeReference(to: newValue)
+        }
+    }
+
+    /// :nodoc:
+    public init(wrappedValue: T?) {
+        guard let wrappedValue = wrappedValue else {
+            self.threadSafeReference = nil
+            self.rlmConfiguration = nil
+            return
+        }
+        guard let config = wrappedValue.realm?.configuration else {
```

Comment:
This should grab the configuration in the same way as the above code (or this should be just `self.wrappedValue = wrappedValue` if that's legal).